wandb/sdk/wandb_init.py

```diff
@@ -837,6 +837,7 @@ def init(
     save_code=None,
     id=None,
     settings: Union[Settings, Dict[str, Any], None] = None,
+    job_name: Optional[str] = None,
```

Comment:
why both here and in the settings?

Reply:
Aren't init args passed to the settings which are then passed to the run object?

Reply:
ah sorry, what i meant is why do we want it as new arguments in `init` and not just part of settings...

Reply:
I think this will make it more accessible for users, when we eventually remove the opt-in nature of producing jobs

Reply:
@KyleGoyette can you link a design doc or some other reference so I can understand how this is being used.

Reply:
https://www.notion.so/W-B-Job-API-33b85fec312f450dbe6235cb685c5fb5#e878c1f7dc2744f287894a3931558b83

Reply:
This PR just lets them decide the name of the output job artifact